File: internal/util/password.go
package util

import (
	"github.com/96Asch/mkvstage-server/internal/domain"
	"golang.org/x/crypto/bcrypt"
)

func Encrypt(password string) (string, error) {
	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
	return string(hash), err
}

func Validate(plaintext, hash string) error {
	err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(plaintext))
	if err != nil {
		return domain.NewInternalErr()
	}

	return nil
}

File: auth/src/util/password.ts
import bcrypt from 'bcrypt';

const saltRounds: number = 10;

const hash = async (password: string): Promise<string> => {
    return bcrypt.hash(password, saltRounds);
};

const validate = async (password: string, hash: string): Promise<boolean> => {
    return bcrypt.compare(password, hash);
};

export default Object.freeze({
    hash,
    validate,
});
